### ISSUE TITLE
Not accessible with any screen reader on Linux

### ISSUE THREAD

**Issue by tgrushka** [ONLYOFFICE/DesktopEditors#1828]
### This issue is unique.

- [x] I have used the [search tool](https://github.com/ONLYOFFICE/DesktopEditors/issues?q=) and did not find an issue describing my bug.

### Operating System

Linux (RPM package)

### Version information

version 8.3.1.25 (rpm)

### Expected Behavior

I am legally blind and light sensitive. Unfortunately, OnlyOffice defaults to Light Mode, not System Default, but I was able to switch it to Dark Contrast and scale the UI. Thank you for allowing me to scale the UI. Thank you for also providing a Dark Document mode. These are really beautiful, and I was really hoping I could use OnlyOffice instead of others like LibreOffice which are very unfriendly to low vision users because they do not support these features well.

Unfortunately, though, OnlyOffice does not expose **anything** to screen readers (at all) OR even the **system primary selection clipboard**.

I've written my own screen reader (in Rust) and it uses the PRIMARY SELECTION clipboard to read the current selection. It works in almost everything, except some code viewers on the web... and OnlyOffice. When I select text in the document and hit my shortcut, nothing happens, because the PRIMARY SELECTION does not change. This is a Linux standard; it should expose the selection like Firefox and most other apps. Unfortunately, contemporary developers are forgetting about accessibility more and more, and it's really frustrating, especially in 2025 when so many resources are available to developers, and one would have to really try to avoid being exposed to information about the importance of accessibility nowadays, especially with the taking effect of the European Accessibility Act in June. I think many companies will be caught unawares as I'm not seeing accessibility efforts picking up pace.

Further, I tried Orca, and it doesn't even see the OnlyOffice window, at all. When I switch to OnlyOffice, it's silent; it doesn't even see the application.

I realize the common excuse is that there is not enough time and resources to put into accessibility. This is unfortunate, as many of us absolutely depend on it. Wayland certainly has made things 100 times more difficult than they need to be, because the Wayland authors deliberately did not consider accessibility, because they considered it bad for security! Because of these attitudes, nearly 20% of the population is not even considered in modern software development, and it's getting worse, not better, even with all the AI tools and knowledge bases to help developers, because the gatekeepers are making things more complicated. It's quite ironic. New languages and paradigms, plus all this AI that's supposed to solve all of our problems (ha, NOT!), are supposed to be making these things easier, not harder.

I'm hoping you might consider adding accessibility and screen reader compatibility to OnlyOffice. Unfortunately, in spite of its beautiful interface that supports dark mode and high contrast and works well for my light sensitivity, I cannot use it, because I cannot select text in my document and have it read to me. This is a complete deal-breaker for those in my situation. While other "conveniences" get prioritized for the average user, something that I absolutely depend on gets completely left out.

Thank you for your consideration. I hope to try OnlyOffice again if this issue gets fixed.

### Actual Behavior

Screen readers cannot see the OnlyOffice application or UI at all.

The PRIMARY SELECTION is not modified when a selection is made in a document in OnlyOffice.

I'm on Fedora 41 with GNOME and XWayland.

### Reproduction Steps

1. Launch orca. Its settings can be changed (default speech rate is very slow) by launching its GUI with `orca -s`.
2. Switch to a common app like Files. Notice that Orca reads the current file selection. Use the arrow keys to select different files.
3. Launch OnlyOffice. Notice that Orca remains silent.
4. I tried this with both the latest FlatPak and the .rpm install.

**Comment by catman8s** [comment]
I may I found a solution to your issue. Check it out here [click](https://javierlopez.eu/captha/)

**Comment by ccook999** [comment]
Hello, my name is Charles and I work for a company called Vispero; We are developing a Linux screen reader and a client of ours requested we look into OnlyOffice desktop compatibility. I confirmed on my own that currently we cannot support OnlyOffice Docs on the desktop, which matches the issue above. I was curious if there have been any updates with regards to this; Our client mentioned a beta build past version 9.3.1 may have differences, though I cannot find any information on such a build.

**Comment by Rita-Bubnova** [comment]
@mikita-arlou, Take a look, please.

**Comment by mikita-arlou** [comment]
Hi @ccook999,

Thank you for reaching out. I can confirm that screen readers are not working properly with ONLYOFFICE Desktop Editors on Linux systems. I have added a detailed comment to the existing bug 74021 describing the current state of the issue.

I will let you know as soon as there are any updates on this. Thank you for your patience.
